dbcon/execplan/simplecolumn.cpp

```diff
@@ -471,22 +471,25 @@ bool SimpleColumn::sameColumn(const ReturnedColumn* rc) const
 
 void SimpleColumn::setDerivedTable()
 {
-    if (hasAggregate())
+    if (hasAggregate() || hasWindowFunc())
     {
         fDerivedTable = "";
         return;
     }
+    ReturnedColumn* rc = dynamic_cast<ReturnedColumn*>(fDerivedRefCol);
```

Comment:
You don't need full-blown dynamic_cast here and simple typeid call would be enough. dynamic_cast traverses the whole tree of the classes hierarchy whilst typeid don't.